modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java

```diff
@@ -59,7 +62,24 @@ public String getDriverClassName() {
     @Override
     public String getJdbcUrl() {
         // Disable Postgres driver use of java.util.logging to reduce noise at startup time
-        return "jdbc:postgresql://" + getContainerIpAddress() + ":" + getMappedPort(POSTGRESQL_PORT) + "/" + databaseName + "?loggerLevel=OFF";
+        return format("jdbc:postgresql://%s:%d/%s?loggerLevel=OFF", getContainerIpAddress(), getMappedPort(POSTGRESQL_PORT), databaseName);
+    }
+
+    @Override
+    protected String constructUrlForConnection(String queryString) {
+        String baseUrl = getJdbcUrl();
+
+        if ("".equals(queryString)) {
+            return baseUrl;
+        }
+
+        if (!queryString.startsWith("?")) {
+            throw new IllegalArgumentException("The '?' character must be included");
+        }
+
+        return baseUrl.contains("?")
```

Comment:
startsWith

Reply:
That's the base jdbc URL and not query string to append.

The logic here is supposed to cater for the situation where the base URL doesn't contain a query string (then we can just concat them instead of rebuilding the query string first)

Reply:
sorry, my bad, was reading from phone.

Reply:
no hard feelings :)